src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -260,7 +259,15 @@ const AppWrapper = (props: { children: ReactNode }) => {
     } else {
       dispatch(setPauseQueueIncomingRequest(true));
     }
-  }, [isOnline, authentication.loggedIn, dispatch]);
+  }, [isOnline, authentication.loggedIn, isMessagesHandled, dispatch]);
+
+  useEffect(() => {
+    PeerConnection.peerConnection.onPeerDisconnectedStateChanged(
+      async (event) => {
+        return peerDisconnectedChangeHandler(event, connectedWallet, dispatch);
+      }
+    );
+  }, [connectedWallet, dispatch]);
```

Comment:
Hmm, I wouldn't have thought this is needed - the `connectedWallet` value doesn't auto-update when `peerDisconnectedChangeHandler` is called?

Reply:
interesting! cool, thanks for fixing :)